src/buckets/components/createBucketForm/UpdateBucketOverlay.tsx

```diff
@@ -151,9 +151,17 @@ const UpdateBucketOverlay: FunctionComponent = () => {
         ],
       })
     } else {
+      if (CLOUD) {
+        setBucketDraft({
+          ...bucketDraft,
+          retentionRules: undefined,
+        })
+        return
+      }
+
       setBucketDraft({
         ...bucketDraft,
-        retentionRules: [],
+        retentionRules: [{everySeconds: 0}],
```

Comment:
I think we can make this a one-line change (and omit the multiple `setBucketDraft` invocations) if we just use a ternary operator here. Would also be safer imho as it's clear from the line itself that there is no change in Cloud environments to the existing behavior.

>      setBucketDraft({
>         ...bucketDraft,
>         retentionRules: CLOUD ? [] : [{everySeconds: 0}],
>       })
>

Reply:
Note also that the above change is difference in that, rather than passing an array, it would pass `undefined`. Let's stick to the existing behavior of an empty array so as to avoid an unintentional change in behavior in C2.